RecoMuon/MuonIdentification/src/MuonShowerInformationFiller.cc

```diff
@@ -871,15 +894,17 @@ void MuonShowerInformationFiller::fillHitsByStation(const reco::Muon& muon) {
       }
 
      //for theta
+     TransientTrackingRecHit::ConstRecHitContainer muonRecHitsThetaBest;
      if (!muonCorrelatedHits.at(stat).empty()) {
 
+       TransientTrackingRecHit::ConstRecHitContainer muonRecHitsThetaTemp;
        float dthetamax = 0;
        for (TransientTrackingRecHit::ConstRecHitContainer::const_iterator iseed = muonCorrelatedHits.at(stat).begin(); iseed != muonCorrelatedHits.at(stat).end(); ++iseed) {
            if (!(*iseed)->isValid()) continue;
            GlobalPoint refpoint = (*iseed)->globalPosition(); //starting from the one with smallest value of phi
            muonRecHitsThetaTemp.clear();
            muonRecHitsThetaTemp = findThetaCluster(muonCorrelatedHits.at(stat), refpoint);
-       }//loop over seeds 
+       }//loop over seeds
        if (muonRecHitsThetaTemp.size() > 1) {
          float dtheta = fabs((float)muonRecHitsThetaTemp.back()->globalPosition().theta() - (float)muonRecHitsThetaTemp.front()->globalPosition().theta());
          if (dtheta > dthetamax) {

```

Comment:
I think that the problem with the 4th station is inadvertently here or on L897:
in station 4 there is no theta measurement; this means that dtheta = 0 for muons  and muonRecHitsThetaBest remains at zero now; while in the previous version of the code it would be the previous station value (which probably makes sense; unless it's accidental).

To keep the same physics performance, please move the muonRecHitsThetaBest back to where it was and let's see if it makes an improvement.